src/docs/release/breaking-changes/use-maxLengthEnforcement-instead-of-maxLengthEnforced.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Use maxLengthEnforcement instead of maxLengthEnforced
+description: Introduing MaxLengthEnforcement enum.
+---
+
+## Summary
+
+_If you didn't use `maxLengthEnforced` in `TextField`/`TextFormField`/`CupertinoTextField`, or you don't care about the truncate behavior for the composing text editing value, you can stop reading._
+
+To control the behavior of `maxLength` in the `LengthLimitingTextInputFormatter`, use `maxLengthEnforcement` instead of `maxLengthEnforced`, and `maxLengthEnforced` is deprecated.
+
+## Context
+
+Before [the first PR][Fix TextField crashed with composing and maxLength set], `LengthLimitingTextInputFormatter` will always truncate the editing value, which may leads to engine crashes when the composing range is overflowed sometimes. With this changes, a composing `TextEditingValue` will not be truncated if only `maxLength` set, and still be truncated after the composing ends (See [`maxLength` documentation][] for more specific behavior). But it's not optional.
+
+With [the second PR][Introduce `MaxLengthEnforcement`], all behaviors of max length limiting will be controlled by passing `maxLenghLimitingEnforcement` to the `LengthLimitingTextInputFormatter`. It's also exposed to `TextField`/`TextFormField`/`CupertinoTextField`. Since then, `maxLengthEnforced` is deprecated. (See also: [`MaxLengthEnforcement` documentation][`MaxLengthEnforcement`] .)
+
+## Description of change
+
+* Add `MaxLenghLimitingEnforcement` enum, which controlls all max length limiting behavious.
+* `maxLengthEnforced` in text fields is deprecated.
+* A composing `TextEditingValue` will skip the `maxLength` limit by default.
+* A named optional parameter `maxLenghLimitingEnforcement` has been added to the `TextField`/`TextFormField`/`CupertinoTextField`/`LengthLimitingTextInputFormatter`, defaults to `MaxLengthLimitingEnforcement.regular`.
+
+## Migration guide
+
+### Not enforces limitation
+
+If you want a max length warning, but not truncate when it's exceed the limitation, use `MaxLengthEnforcement.warningOnly` instead of `maxLengthEnforced: false`.
```

Comment:
```
To print a max length warning, but _not_ truncate when the limit is exceeded,
use `MaxLengthEnforcement.warningOnly`, instead of `maxLengthEnforced: false`.

```